### PR TITLE
71 get token historic price data

### PR DESCRIPTION
**Pull request by verschhh**
feel free to refer to this for the **Definition of done**

### CODE REVIEW

File: src/alchemy/alchemy.service.ts [hunk 1]
```diff
@@ -111,4 +111,69 @@ export class AlchemyService {
...
+                console.error(`API returned an error:`, error.response.data);
+            } else if (error.request) {
+                console.error(`No response received from API:`, error.request);
+            } else {
+                console.error(`Error setting up request:`, error.message);
+            }
+            throw error;
+        }
+    }
+}
```
Comment (PaulThblt) [inline src/alchemy/alchemy.service.ts]:
```suggestion
}

```

File: src/app.module.ts [hunk 2]
```diff
@@ -13,4 +13,4 @@ import { AlchemyModule } from './alchemy/alchemy.module';
   controllers: [AppController],
   providers: [AppService],
 })
-export class AppModule {}
+export class AppModule {}
```
Comment (PaulThblt) [inline src/app.module.ts:16]:
```suggestion
export class AppModule {}

```

File: src/alchemy/alchemy.controller.ts [hunk 3]
```diff
@@ -9,4 +9,15 @@ export class AlchemyController {
...
     }
+    @Get('token-price-in-euro/:token')
+    async getTokenPriceInEuro(@Param('token') token: string): Promise<any> {
+        return this.alchemyService.getTokenPriceInEuro(token);
+    }
+    @Post('get-token-history/:token')
+    async getTokenHistory(@Param('tokenAdress') token: string, @Body() body: { beginDate: string; endDate: string; interval: Number; tokenAddress: string; network: string}): Promise<any> {
+        const beginDate = new Date(body.beginDate);
+        const endDate = new Date(body.endDate);
+        console.log(body.tokenAddress, body.network);
```
Comment (PaulThblt) [inline src/alchemy/alchemy.controller.ts]: is the log necessary here?

File: src/alchemy/alchemy.service.ts [hunk 4]
```diff
@@ -111,4 +111,69 @@ export class AlchemyService {
...
+            }
+        }
+        return results;
+    }
+
+    async getTokenHistoricPrice(symbol: string, beginDate: Date, endDate: Date, interval: Number, tokenAddress: string, network: string): Promise<any> {
+        const formattedBeginDate = beginDate.toISOString();
+        const formattedEndDate = endDate.toISOString();
+        const url = `https://api.g.alchemy.com/prices/v1/${this.apiKey}/tokens/historical`;
+        console.log(tokenAddress, network);
```
Comment (PaulThblt) [inline src/alchemy/alchemy.service.ts:163]: Same here

Comment (PaulThblt) [reply src/alchemy/alchemy.service.ts:163]: and below in the file

**Review by verschhh — COMMENTED** [review]
The change have been applied, thanks for the feedback @PaulThblt !

**Review by KentinPaille — APPROVED** [review]

**Pull request merged**